rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala

```diff
@@ -65,7 +65,17 @@ case class NodeConfigurationCache(
   , nodeContextCache: Int
   , policyCache: Set[PolicyCache]
 
-)
+) {
+  // We need a method to correctly compare a NodeConfigurationCache that was serialized
+  // from a NodeconfigurationCache created from a NodeConfiguration (so without writtenDate)
+  def equalWithoutWrittenDate(other: NodeConfigurationCache) : Boolean = {
```

Comment:
NITPICKING time: 

What do you think of the solutions from this thread ? (need @fanf eye on this too!) 

http://stackoverflow.com/questions/10373715/scala-ignore-case-class-field-for-equals-hascode

I like the first solution (having case class NodeConfigurationCache(id : NodeId, nodeInfo : ...)(val writtenDate : ...) but it would prevent comparing on writtenDate when it is defined, so we may want to exclude this 

Maybe rewriting equals and checking None cases would be better ? 


Maybe we should have transform all our NodeConfigurationCache into NodeConfiguration and compare them instead of comparing NodeConfigurationCache ?